Install-DVWA.sh

```diff
@@ -0,0 +1,174 @@
+        success_message=$(get_language_message "\033[92m$1 is installed!\033[0m" "\033[92m$1 !Está instalado!\033[0m")
+        echo -e "$success_message"
+    fi
+}
+
+# MySQL root password prompt function
+get_mysql_root_password() {
+    read -s -p "$(get_language_message "Enter MySQL root password: " "Ingrese la contraseña de root de MySQL: ")" mysql_root_password
+    echo -e "$mysql_root_password"
+}
+# ASCII Art
+echo -e "\033[96m\033[1m
+                  ██████╗ ██╗   ██╗██╗    ██╗ █████╗                    
+                  ██╔══██╗██║   ██║██║    ██║██╔══██╗                   
+                  ██║  ██║██║   ██║██║ █╗ ██║███████║                   
+                  ██║  ██║╚██╗ ██╔╝██║███╗██║██╔══██║                   
+                  ██████╔╝ ╚████╔╝ ╚███╔███╔╝██║  ██║                   
+                  ╚═════╝   ╚═══╝   ╚══╝╚══╝ ╚═╝  ╚═╝                   
+                                                                        
+  ██╗███╗   ██╗███████╗████████╗ █████╗ ██╗     ██╗     ███████╗██████╗ 
+  ██║████╗  ██║██╔════╝╚══██╔══╝██╔══██╗██║     ██║     ██╔════╝██╔══██╗
+  ██║██╔██╗ ██║███████╗   ██║   ███████║██║     ██║     █████╗  ██████╔╝
+  ██║██║╚██╗██║╚════██║   ██║   ██╔══██║██║     ██║     ██╔══╝  ██╔══██╗
+  ██║██║ ╚████║███████║   ██║   ██║  ██║███████╗███████╗███████╗██║  ██║
+  ╚═╝╚═╝  ╚═══╝╚══════╝   ╚═╝   ╚═╝  ╚═╝╚══════╝╚══════╝╚══════╝╚═╝  ╚═╝                                                                        
+\033[0m"
+
+welcome_message=$(get_language_message "\033[96mWelcome to the DVWA installer!\033[0m" "\033[96m¡Bienvenido al instalador de DVWA!\033[0m")
+echo -e "$welcome_message"
+
+echo
+# Start of the installer
+
+# Update the repositories
+update_message=$(get_language_message "\e[96mUpdating repositories...\e[0m" "\e[96mActualizando repositorios...\e[0m")
+echo -e "$update_message"
+apt update
+
+# Chek if the dependencies are installed
+dependencies_message=$(get_language_message "\e[96mVerifying and installing necessary dependencies...\e[0m" "\e[96mVerificando e instalando dependencias necesarias...\e[0m")
+echo -e "$dependencies_message"
+
+check_program apache2
+check_program mariadb-server
+check_program mariadb-client
+check_program php
+check_program php-mysqli
+check_program php-gd
+check_program libapache2-mod-php
+check_program git
+
+# Download the repo
+download_message=$(get_language_message "\e[96mDownloading DVWA from GitHub...\e[0m" "\e[96mDescargando DVWA desde GitHub...\e[0m")
+echo -e "$download_message"
+git clone https://github.com/digininja/DVWA.git /var/www/html/DVWA
+sleep 2
+
+# Start MySql
+mysql_start_message=$(get_language_message "\e[96mStarting MySQL...\e[0m" "\e[96mIniciando MySQL...\e[0m")
+echo -e "$mysql_start_message"
+systemctl start mysql.service
+sleep 2
+
+# User Prompt for the MySQL root password
+mysql_root_password=$(get_mysql_root_password)
+
+# Run MySQL commands
+mysql -u root -p$mysql_root_password -e "CREATE DATABASE IF NOT EXISTS dvwa;"
```

Comment:
If the password is blank then this will force the user to press enter each time I think. Also, if the password has special characters - quotes, pipes etc - in it it will probably break the command.

Reply:
Solved, I added the option to press enter to use a blank password and added "" in the -p "$mysql_root_password" so that this prevents the use of commands through the script and make sure the special characters in the password.

Reply:
have you checked to see what happens if there is a double quote in the password?

Reply:
I think it should be ok now, I have redone the function for the mysql connection, now it should do the job if it has special characters or not and if the user provides password or not.

run_mysql_commands() {
    local mysql_user
    local mysql_password

    read -p "$(get_language_message "\e[96mEnter MySQL user:\e[0m " "\e[96mIngrese el usuario de MySQL:\e[0m ")" mysql_user
    read -s -p "$(get_language_message "\e[96mEnter MySQL password (press Enter for no password):\e[96m " "\e[96mIngrese la contraseña de MySQL (presiona Enter si no hay contraseña):\e[0m ")" mysql_password
    echo -e "\n$(get_language_message "\e[96mCredentials provided.\e[0m" "\e[96mCredenciales proporcionadas.\e[0m")"

    # Check if a password was provided
    if [ -n "$mysql_password" ]; then
        # Execute MySQL commands with password
        mysql -u "$mysql_user" -p"$mysql_password" -e "CREATE DATABASE IF NOT EXISTS dvwa;" &>/dev/null &&
        mysql -u "$mysql_user" -p"$mysql_password" -e "CREATE USER 'dvwa'@'localhost' IDENTIFIED BY 'p@ssw0rd';" &>/dev/null &&
        mysql -u "$mysql_user" -p"$mysql_password" -e "GRANT ALL PRIVILEGES ON dvwa.* TO 'dvwa'@'localhost';" &>/dev/null &&
        mysql -u "$mysql_user" -p"$mysql_password" -e "FLUSH PRIVILEGES;" &>/dev/null
    else
        # Execute MySQL commands without password
        mysql -u "$mysql_user" -e "CREATE DATABASE IF NOT EXISTS dvwa;" &>/dev/null &&
        mysql -u "$mysql_user" -e "CREATE USER 'dvwa'@'localhost' IDENTIFIED BY 'p@ssw0rd';" &>/dev/null &&
        mysql -u "$mysql_user" -e "GRANT ALL PRIVILEGES ON dvwa.* TO 'dvwa'@'localhost';" &>/dev/null &&
        mysql -u "$mysql_user" -e "FLUSH PRIVILEGES;" &>/dev/null
    fi

    if [ $? -eq 0 ]; then
        echo "$(get_language_message "\033[92mMySQL commands executed successfully.\033[0m" "\033[92mComandos MySQL ejecutados con éxito.\033[0m")"
    else
        echo -e "$(get_language_message "\033[91mError: Unable to execute MySQL commands. Please check your MySQL credentials." "\033[91mError: No se pueden ejecutar los comandos de MySQL. Por favor, verifique sus credenciales de MySQL.")"
    fi
}